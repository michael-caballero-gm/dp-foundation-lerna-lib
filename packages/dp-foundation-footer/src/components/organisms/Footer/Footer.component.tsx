import React from 'react';
import { StyledFooter } from './Footer.styled';
import { Link } from '@greatminds/dp-atomic-ui-lib';

interface IFooter {
  /**
   * Defines if the footer is used in habitat apps or not
   */
  isHabitatContext?: boolean;
}

export const Footer: React.FC<IFooter> = ({isHabitatContext = true}) => {
  return(
    // StyledFooter was stolen from Cortex, DO NOT leave it like this, implement own styles
    <StyledFooter>
      {isHabitatContext ? <span>The footer should be clear</span> : <span>The footer should be dark</span>}<br/>
      here we could reuse Cortex components like <Link href="https://www.google.com" target="_blank">Link</Link>  or make our own implementation
    </StyledFooter>
  );
};

// export default Footer;
