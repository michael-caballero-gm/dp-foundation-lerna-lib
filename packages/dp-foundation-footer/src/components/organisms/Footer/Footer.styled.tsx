import styled, { css, ThemeProps } from 'styled-components';
import { BaseTheme } from '~/themes/baseTheme';

export const StyledFooter = styled.footer(
  ({ theme: { colors, typography, measurements } }: ThemeProps<BaseTheme>) => css`
    background-color: ${colors.beige[60]};
    box-sizing: border-box;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 56px;
    min-height: 48px;
    width: 100%;
    border-top: 1px solid ${colors.beige[140]};

    a,
    p {
      font-size: ${typography.sizes.extraSmall};
      text-transform: uppercase;
      line-height: 1.333;
      letter-spacing: 1px;
      font-family: ${typography.fonts.primary};
    }

    .links {
      display: flex;
      a {
        text-decoration: none;
        margin-right: calc(${measurements.extraLarge} * 2);
      }
    }

    .copyright {
      display: flex;
      p {
        color: ${colors.n[100]};

        &:not(:first-child):before {
          content: '|';
          margin: 0 5px;
        }
      }
    }
  `,
);
