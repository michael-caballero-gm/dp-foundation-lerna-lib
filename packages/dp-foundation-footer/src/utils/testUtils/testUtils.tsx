import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '@theme/baseTheme';

const Providers = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

const customRender = (ui, options?): RenderResult => render(ui, { wrapper: Providers, ...options });

/**
 * Window.location.href assignation for navigation triggers error when trying to execute, calling this function at the begining of your test file
 * prevents the error to happen
 */
export const preventWindowNavigationError = () => {
  delete window.location;
  window.location = {
    href: '',
  } as any;
};

/**
 * Injects in the global scope an implementation of a given hook returning the expected fake value
 * it is required to add jest.mock('~/lib/hooks'); at the beginning of your .spec file
 * @param mockHooks List of Hooks required to be mocked by a given test case
 */
export const mockCustomHooks = (mockHooks: Array<{ hook; returnedValue }>) => {
  mockHooks.forEach(({ hook, returnedValue }) => {
    hook.mockReturnValue(returnedValue);
  });
};

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';

// override render method
// eslint-disable-next-line import/export
export { customRender as render };
