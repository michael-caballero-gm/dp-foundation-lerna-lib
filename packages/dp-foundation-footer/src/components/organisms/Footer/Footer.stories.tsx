import React from 'react';
import styled from 'styled-components';

import { Footer } from './Footer.component';

const StyledContainer = styled.div`
  > div {
    margin-left: 10px;
  }
`;

/* Storybook Footer */
export default {
  title: 'Organisms/Footer',
  component: Footer,
};

export const Default = (props) => <Footer {...props}/>;
