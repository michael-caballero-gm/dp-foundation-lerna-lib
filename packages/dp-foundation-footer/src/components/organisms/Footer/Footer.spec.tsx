import 'jest-styled-components';
import { render, cleanup } from '@utils/testUtils/testUtils';
import { Footer } from './Footer.component';

afterEach(cleanup);
describe('Footer', () => {

  it('should render correctly', () => {
    const { container, debug } = render(<Footer />);
    debug();
    expect(container).toMatchSnapshot();
  });
});
