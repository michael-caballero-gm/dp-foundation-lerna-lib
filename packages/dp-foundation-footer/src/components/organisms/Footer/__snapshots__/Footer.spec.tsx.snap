// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
.c0 {
  cursor: pointer;
  color: #407177;
  font-family: Circular std;
  font-weight: 400;
  font-size: 0.875rem;
}

.c0:hover {
  color: #649295;
}

.c0:active {
  color: #7CA3A1;
}

.c0:focus {
  border: 3px solid #7A7562;
  border-radius: 4px;
}

<div>
  <div>
    <span>
      The footer should be clear
    </span>
    <br />
    here we could reuse Cortex components like 
    <a
      class="c0"
      href="https://www.google.com"
      target="_blank"
    >
      Link
    </a>
      or make our own implementation
  </div>
</div>
`;
